lib/spack/spack/test/directives.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import spack.repo
+from spack.spec import Spec
+
+
+def test_false_directives_do_not_exist(mock_packages):
+    cls = spack.repo.path.get_pkg_class('when-directives-false')
+    assert not cls.dependencies
+    assert not cls.resources
+    assert not cls.patches
+
+
+def test_true_directives_exist(mock_packages):
```

Comment:
(minor) Maybe briefly explain a "true directive": e.g. `a directive with a when condition that evaluates to `True` at import time`.